charts/kyverno/templates/hooks/clusterrole.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "kyverno.pre-upgrade-job.name" . }}
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "-10"
+rules:
+- apiGroups: [""]
+  resourceNames:
+  - {{ include "kyverno.pre-upgrade-job.name" . }}
+  resources: [podsecuritypolicies]
+  verbs: [get, list, patch, create, delete, use]
```

Comment:
PSP references need to be wrapped in a conditional so they won't get deployed on CSM 1.7 where they don't exist.  Use:
```
{{- if .Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy" }}
...
{{- end}}```

Reply:
Wait, if you just remove this, will this still work on CSM 1.6 where PSP is being used with Kubernetes 1.24?

Reply:
The changes we are submitting is for a scenario of upgrading from CSM 1.6 to CSM 1.7 where PSPs are not there.